kappuccino/src/main/kotlin/br/com/concretesolutions/kappuccino/custom/menu/MenuMethods.kt

```diff
@@ -0,0 +1,60 @@
+package br.com.concretesolutions.kappuccino.custom.menu
+
+import android.support.annotation.IdRes
+import android.support.annotation.StringRes
+import android.support.test.InstrumentationRegistry.getTargetContext
+import android.support.test.espresso.Espresso.onView
+import android.support.test.espresso.Espresso.openActionBarOverflowOrOptionsMenu
+import android.support.test.espresso.ViewInteraction
+import android.support.test.espresso.action.ViewActions
+import android.support.test.espresso.matcher.ViewMatchers.withId
+import android.support.test.espresso.matcher.ViewMatchers.withText
+
+fun menu(openOptionsMenu: Boolean = true, func: MenuMethods.() -> MenuMethods): MenuMethods {
+    return MenuMethods().apply {
+        if (openOptionsMenu)
+            openOptionsMenu()
+        func()
+    }
+}
+
+class MenuMethods {
+
+    fun openOptionsMenu(): MenuMethods {
+        openActionBarOverflowOrOptionsMenu(getTargetContext())
+        return this
+    }
+
+    fun onItem(@StringRes itemTextId: Int, func: Interactions.() -> Interactions): MenuMethods {
```

Comment:
Why did you decided to match the views using `withText()`?
Maybe it's a good idea to let the user match the view with some other attribute like the `Id`.

Reply:
Unfortunately options menu items are only matched with their text (I tried with the ID and it didn't work), only action bar items seem to work that way

Reply:
This is really weird! haha but I think that there's not much to do then... 🤷🏽‍♂️

Reply:
Yeah I know =/